client/my-sites/plans-grid/components/features-grid/index.tsx

```diff
@@ -171,13 +174,16 @@ class FeaturesGrid extends Component< FeaturesGridProps > {
 					'plan-features-2023-grid__mobile-plan-card',
 					getPlanClass( gridPlan.planSlug )
 				);
+
+				const isNotFreePlan = ! isFreePlan( gridPlan.planSlug );
+
 				const planCardJsx = (
 					<div className={ planCardClasses } key={ `${ gridPlan.planSlug }-${ index }` }>
 						{ this.renderPlanLogos( [ gridPlan ] ) }
 						{ this.renderPlanHeaders( [ gridPlan ] ) }
-						{ this.renderPlanTagline( [ gridPlan ] ) }
-						{ this.renderPlanPrice( [ gridPlan ] ) }
-						{ this.renderBillingTimeframe( [ gridPlan ] ) }
+						{ isNotFreePlan && isInSignup && this.renderPlanTagline( [ gridPlan ] ) }
+						{ isNotFreePlan && this.renderPlanPrice( [ gridPlan ] ) }
+						{ isNotFreePlan && this.renderBillingTimeframe( [ gridPlan ] ) }
```

Comment:
I have a concern with the use of `isInSignup` here. When we render the plans-grid in another context, we'd probably want it to have it behave like in Signup, and a logged-out page won't get `isInSignup` prop, as it won't be signup. So we'd want to keep adding `isInSignup II isInLoggedOut`, etc. (which will start to look cluttered)

Is there a way to avoid using the `isInSignup` prop here? Otherwise, it looks like the exception for this case would be `isCalypsoAdmin` instead. Maybe we can introduce that prop already and replace this condition with `! isCalypsoAdmin` 🤔

Reply:
Or if you can think of a different/better abstraction or way..